PCbuild/find_python.bat

```diff
@@ -34,10 +34,10 @@
 @if exist "%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe" (set PYTHON="%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe") & (set _Py_Python_Source=found in externals directory) & goto :found
 
 @rem If HOST_PYTHON is recent enough, use that
-@if NOT "%HOST_PYTHON%"=="" @%HOST_PYTHON% -Ec "import sys; assert sys.version_info[:2] >= (3, 6)" >nul 2>nul && (set PYTHON="%HOST_PYTHON%") && (set _Py_Python_Source=found as HOST_PYTHON) && goto :found
+@if NOT "%HOST_PYTHON%"=="" @%HOST_PYTHON% -Ec "import sys; assert sys.version_info[:2] >= (3, 8)" >nul 2>nul && (set PYTHON="%HOST_PYTHON%") && (set _Py_Python_Source=found as HOST_PYTHON) && goto :found
 
 @rem If py.exe finds a recent enough version, use that one
-@for %%p in (3.8 3.7 3.6) do @py -%%p -EV >nul 2>&1 && (set PYTHON=py -%%p) && (set _Py_Python_Source=found %%p with py.exe) && goto :found
+for /F "tokens=2 delims=-" %%p in ('py -0 2^>nul') do @py -%%p -Ec "import sys; assert sys.version_info[:2] >= (3, 8)" >nul 2>&1 && (set PYTHON=py -%%p) && (set _Py_Python_Source=found %%p with py.exe) && goto :found
```

Comment:
Note that `py` doesn't actually sort versions before displaying them, and currently 3.10 sorts after all the rest.

I'd rather keep this in the same format as before, and just update the version numbers.

```suggestion
@for %%p in (3.9 3.8) do @py -%%p -EV >nul 2>&1 && (set PYTHON=py -%%p) && (set _Py_Python_Source=found %%p with py.exe) && goto :found
```

Reply:
> Note that py doesn't actually sort versions before displaying them, and currently 3.10 sorts after all the rest.

Thanks for the review, hmm I thought this was solved in https://bugs.python.org/issue38506? And even if the order is wrong for older versions of the ``py`` launcher, the ``import sys; assert sys.version_info[:2] >= (3, 8)`` guards against using a wrong version.

I don't have any strong feelings towards any approach. So feel free to commit your suggestion if you want to :).

Reply:
Yeah, you should be right, so I wonder why I don't have that version installed...

In any case, I still prefer the static list of versions to look for, as it'll reduce the likelihood of random build breaks on machines.